msg.go

```diff
@@ -732,12 +732,9 @@ func (dns *Msg) PackBuffer(buf []byte) (msg []byte, err error) {
 
 	// We need the uncompressed length here, because we first pack it and then compress it.
 	msg = buf
-	compress := dns.Compress
```

Comment:
[ Quoting <notifications@github.com> in "Re: [miekg/dns] don't modify Compre..." ]
>> @@ -732,12 +732,9 @@ func (dns *Msg) PackBuffer(buf []byte) (msg []byte, err error) {
>>
>>  	// We need the uncompressed length here, because we first pack it and then compress it.
>>  	msg = buf
>> -	compress := dns.Compress
>
>it's possible. let's say you implement a caching DNS server that caches a DNS response, and you serve up two requests for the same domain (which has been cached) at roughly the same time.

But then you would have a problem on the query ID as well? Asked diff.: how do
fix the query id there?

/Miek

--
Miek Gieben